ui/ui-components/client/client.ts

```diff
@@ -103,161 +43,92 @@ export class Client {
     } = () => ({})
   ) {}
 
-  private redirectOnAccessError({ code }: { code: string }): boolean {
-    const { type, req, res } = this.getRequestContext();
-    if (
-      !isBrowser() &&
-      (!req || !res || type === "GetServerSidePropsContext")
-    ) {
-      return false;
+  public async request<Response = any>(
+    verb: Method = "get",
+    path: string,
+    data: AxiosRequestConfig["data"] = {},
+    options: Partial<ClientRequestOptions> = {}
+  ): Promise<Response> {
+    options = { useCache: true, ...options };
+    const headers: AxiosRequestHeaders = { ...DEFAULT_HEADERS };
+
+    const { req } = this.getRequestContext();
+    if (req?.headers?.cookie) {
+      headers["X-GROUPAROO-SERVER-TOKEN"] = SERVER_TOKEN;
+      headers["cookie"] = req?.headers?.cookie;
+      options.useCache = false; // do not ever cache responses on the server
     }
 
-    const redirect = getRedirectFromErrorCode(
-      code,
-      isBrowser() ? window.location.pathname : req.url.match("^[^?]*")[0]
-    );
+    const config: AxiosRequestConfig = {
+      url: BASE_URL + path,
+      method: verb.toLowerCase() as Method,
+      withCredentials: true,
+      headers,
+    };
 
-    if (redirect) {
-      if (isBrowser()) {
-        window.location.href = redirect.destination;
-      } else {
-        res.writeHead(302, {
-          Location: redirect.destination,
-        });
-        res.end();
+    if (data) {
+      data.csrfToken = getCsrfToken();
+
+      for (const i in data) {
+        if (data[i] === null || data[i] === undefined) {
+          delete data[i];
+        }
       }
-    }
 
-    return !!redirect;
-  }
+      config[config.method === "get" ? "params" : "data"] = data;
+    }
 
-  private csrfToken = () => {
-    if (globalThis?.localStorage) {
-      return window.localStorage.getItem("session:csrfToken");
+    let unlock: ClientCacheGetObject["unlock"];
+    if (config.method === "get" && options.useCache) {
+      const { cacheData, unlock: _unlock } = await this.cache.get<Response>(
+        config.url + JSON.stringify(data)
```

Comment:
I think this could potentially cause problems without a stable stringify method.

Reply:
@krishnaglick Can you explain the reasoning? This has also been the same way before I moved the file.